### ISSUE TITLE
NUnit1001 False Positive with params in the list

### ISSUE THREAD

**Issue by jcurl** [nunit/nunit.analyzers#169]
## Version
Testing with NUnit.Analyzers.0.1.0-dev-00216, 
NUnit 2.6.7

## Related Issues
https://github.com/nunit/nunit.analyzers/issues/162

## Problem
There is still a false positive when using the following code snippet

```csharp
[TestCase(1, "str", 2, 1, new string[] { "val", "raw", null, "dlt" })]
[TestCase(2, "str", 2, 1, new string[] { "val", "raw", ".", null })]
[TestCase(3, "str", 2, 1, new string[] { "val", null, ".", "dlt" })]
[TestCase(4, "str", 2, 1, new string[] { null, "raw", ".", "dlt" })]
[TestCase(5, "str", 0, 1, new string[] { "val", "raw", ".", "dlt" })]
[TestCase(6, "str", 2, 0, new string[] { "val", "raw", ".", "dlt" })]
[TestCase(7, null, 0, 0, new string[] { null, null, null, null })]
public void VerifyMissplacedOption(int configNumber, string deviceStr, int expectedTrFileCount, int expectedTrDatabaseCount, params string[] expectedSingleOptions)
{
    ...
}
```

Each of these test cases shows the error:
Error	NUnit1001	The value of the argument at position '4' cannot be assigned to the argument 'expectedSingleOptions'.	IocTraceConverterTest

## Workaround

Remove the `params` keyword.

**Comment by jcurl** [comment]
For me, this is an interesting case. I realize that the workaround is very easy, I can just remove `params` and it works. It tells me that this could potentially be a new analyzer that raises a warning to indicate one shouldn't use params?

**Comment by mikkelbu** [comment]
I think I would rather have that the existing analyzer just handle the `params`. Also perhaps we should write out the types, as this could help us diagnose the problem.
